src/diagnostics.rs

```diff
@@ -2,9 +2,20 @@
 
 #[cfg(feature = "rustc")]
 use cargo_metadata::diagnostic::DiagnosticLevel;
+use std::path::Path;
+
+#[cfg(feature = "rustc")]
+mod rustc_stderr;
+#[cfg(feature = "rustc")]
+pub use rustc_stderr::{cargo_diagnostics_extractor, rustc_diagnostics_extractor};
```

Comment:
I think we can simplify this a bit further by 
* renaming the module to `rustc`
* making it public instead of reexporting things
* moving the cfged impl into it, too